dandiapi/api/views/version.py

```diff
@@ -116,6 +116,11 @@ def update(self, request, **kwargs):
                 locked_version.status = Version.Status.PENDING
                 locked_version.save()
 
+                audit_record = AuditRecord.update_metadata(
+                    dandiset=locked_version.dandiset, user=request.user, metadata=new_metadata
+                )
+                audit_record.save()
```

Comment:
This is yet another instance where our handling of metadata is biting us. In this case, `name` is the only other field included in this endpoint besides `metadata`, even though it is also included in the metadata, and stripped off as a part of processing the request. If by "hidden" fields you're referring to fields which are stripped away and ignored as a part of the metadata update, I don't think we should include them. `name` is a special case which we handle, but those other fields aren't, and including them would just be inconsistent and confusing.

What probably makes the most sense for now is to include `name` alongside the `metadata` field in that record's `details` field. Aside from that, I don't think we should process changed metadata in any special way. I think the way you have it instrumented is fine.

Reply:
Yeah it seems like we could just have `name` be a property derived from the metadata, instead of a seperate field.

```python
@property
def name(self):
    return self.metadata['name']
```

That's another discussion though.